app/controllers/projects_controller.rb

```diff
@@ -0,0 +1,41 @@
+class ProjectsController < ApplicationController
+
+	MAX_VOTES = 3
+
+	def create
+		@hackday = Hackday.find_by_id(params[:project][:hackday_id].to_i)
+		@project = @hackday.projects.build(project_params)
+		if @project.save
+			flash.now[:success] = "Project created!"
+		else
+			flash.now[:danger] = "Project failed to create"
+		end
+		redirect_to @hackday
+	end
+
+	def update
+		@project = Project.find_by_id(params[:id].to_i) 
+		@hackday = @project.hackday
+
+		if @hackday.closed?
+				flash[:danger] = "Hackday has already closed"
+		else
+			current_votes = cookies[:"current_votes_#{@project.hackday_id}"].to_i
```

Comment:
use sessions not cookies.

sessions are encrypted by rails and you can change session stores etc. cookies are a bit more raw than necessary.